frontend/src/pages/admin/FlowBlueprints/dialogs/ImportFlowBlueprintsDialog.vue

```diff
@@ -0,0 +1,241 @@
+                        rows="10"
+                        placeholder="Paste in a collection of blueprints here, in a JSON array."
+                    />
+                </template>
+            </FormRow>
+        </section>
+    </ff-dialog>
+</template>
+
+<script>
+
+import { DocumentAddIcon, DocumentIcon, XIcon } from '@heroicons/vue/outline'
+
+import FormRow from '../../../../components/FormRow.vue'
+
+export default {
+    name: 'ImportFlowBlueprintsDialog',
+    components: { FormRow, DocumentAddIcon, DocumentIcon, XIcon },
+    emits: ['import-blueprints'],
+    setup () {
+        return {
+            show () {
+                this.$refs.dialog.show()
+            }
+        }
+    },
+    data () {
+        return {
+            input: null,
+            file: null,
+            fileOutput: null,
+            errors: {
+                input: null,
+                file: null
+            }
+        }
+    },
+    computed: {
+        jsonInput () {
+            if (this.input) {
+                try {
+                    return JSON.parse(this.input)
+                } catch (err) {
+                }
+            }
+
+            return null
+        },
+        disablePrimaryButton () {
+            if ((this.input && this.errors.input) || (this.file && this.errors.file)) {
+                return true
+            }
+
+            return false
+        }
+    },
+    watch: {
+        input (newVal) {
+            if (!newVal) {
+                this.errors.input = null
+            }
+
+            try {
+                JSON.parse(newVal)
+                this.errors.input = null
+            } catch (err) {
+                this.errors.input = 'Invalid JSON'
+            }
+
+            this.file = null
+        },
+        file (newVal) {
+            if (!newVal) {
+                this.errors.file = null
+            }
+            this.input = null
+        }
+    },
+    methods: {
+        onFileChange (event) {
+            this.file = event.target && event.target.files && event.target.files[0]
+
+            const reader = new FileReader()
+
+            reader.onload = (e) => {
+                try {
+                    this.fileOutput = JSON.parse(e.target.result)
+                    this.errors.file = null
+                } catch (err) {
+                    this.errors.file = 'Invalid JSON'
+                }
+            }
+            reader.readAsText(this.file)
+        },
+        clearInput () {
+            this.input = null
+        },
+        clearModal () {
+            this.input = null
+            this.file = null
+        },
+        onConfirm () {
+            if (this.errors.input || this.errors.file) {
+                return false
+            }
+
+            const bp = this.fileOutput || this.jsonInput
+            this.$emit('import-blueprints', bp)
+            this.$refs.dialog.close()
+            this.clearModal()
+        }
+    }
+}
+</script>
+
+<style lang="scss">
```

Comment:
I'm altering child components styling in this case, scoping the attr would prevent me in doing so. I've made the rules as specific as possible to avoid any side effects